app/controllers/core_induction_programmes/lesson_parts_controller.rb

```diff
@@ -57,12 +57,36 @@ def destroy
     redirect_to lesson_path(lesson)
   end
 
+  def update_progress
+    redirect_to :show and return unless current_user&.early_career_teacher?
+
+    @course_lesson_part = CourseLessonPart.find(params[:lesson_part_id])
+    authorize @course_lesson_part.course_lesson
+    load_progress
+    if @lesson_progress.update(lesson_progress_params)
+      redirect_to module_path(@course_lesson_part.course_lesson.course_module)
+    else
+      render :show
+    end
+  end
+
 private
 
   def load_course_lesson_part
     @course_lesson_part = CourseLessonPart.find(params[:id] || params[:lesson_part_id])
     authorize @course_lesson_part
     @course_lesson_part.assign_attributes(course_lesson_part_params)
+    if current_user&.early_career_teacher?
+      load_progress
+    end
+  end
+
+  def load_progress
+    @lesson_progress = CourseLessonProgress.find_or_create_by!(
+      early_career_teacher_profile: current_user.early_career_teacher_profile,
+      course_lesson: @course_lesson_part.course_lesson,
+    )
+    @lesson_progress.progress = nil
```

Comment:
why do we set this to `nil`? don't we want to load what was persisted in the database?

Reply:
Yes, as far as I can tell. We don't really expect the user to give us the same answer many times, or re-visit the page once it was completed.

I guess maybe if the progress is 'complete' we would want to keep it, but that's probably something to do user research on.